substrate/frame/Cargo.toml

```
@@ -48,6 +48,8 @@ sp-session = { optional = true, workspace = true }
 sp-storage = { optional = true, workspace = true }
 sp-transaction-pool = { optional = true, workspace = true }
 sp-version = { optional = true, workspace = true }
+sp-staking = { features = ["serde"], workspace = true }
+sp-state-machine = { workspace = true }
```

```suggestion
sp-staking = { optional = true, workspace = true }
sp-state-machine = { optional = true, workspace = true }
```

looking at other deps, it makes more sense to have it optional.

How should we do for the serde dependency though? Should we make a new feature `serde` in frame crate? @kianenigma